docs/feature_oled.md

```diff
@@ -0,0 +1,223 @@
+# OLED
+
+## OLED Supported Hardware
+
+128x32 OLED modules using SSD1306 driver IC over I2C. Supported on both AVR and ARM based keyboards. Possible but untested hardware includes other sized OLED modules using SSD1306 over I2C, such as 128x64. 
+
+## Usage
+
+To enable the OLED feature, there are three steps. First, when compiling your keyboard, you'll need to set `OLED_ENABLE=yes` in `rules.mk`, e.g.:
+
+```
+BOOTMAGIC_ENABLE = no
+MOUSEKEY_ENABLE = no
+STENO_ENABLE = no
+EXTRAKEY_ENABLE = yes
+OLED_ENABLE = yes
+```
+
+This enables the feature and the `OLED_ENABLE` define. Then in your `keymap.c` file, you will need to implement the initialization and task calls, e.g:
+
+```C++
+void matrix_init_user(void) {
+#ifdef OLED_ENABLE
+  oled_init(false);   // turns on the display
+#endif
+}
+
+void matrix_scan_user(void) {
+#ifdef OLED_ENABLE
+  oled_task();
+#endif
+}
+```
+
+Then finally implement the user task call and send your data to the oled to render, e.g:
+```C++
+#ifdef OLED_ENABLE
+void oled_task_user(void) {
+  oled_write_P(PSTR("Layer: "), false);
+  switch (biton32(layer_state)) {
+    case _QWERTY:
+      oled_write_P(PSTR("Default\n"), false);
+      break;
+    case _FN:
+      oled_write_P(PSTR("FN\n"), false);
+      break;
+    case _ADJ:
+      oled_write_P(PSTR("ADJ\n"), false);
+      break;
+    default:
+    oled_write_P(PSTR("Undefined\n"), false);
+  }
+
+  // Host Keyboard LED Status
+  uint8_t led_usb_state = host_keyboard_leds();
+  oled_write_P(led_usb_state & (1<<USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
+  oled_write_P(led_usb_state & (1<<USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
+  oled_write_P(led_usb_state & (1<<USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
+}
+#endif
+```
+
+
+## Other Examples
+
+In split keyboards, it is very common to have two oled displays that each render different content. You can do this by switching which content to render by using the return from `has_usb()`, e.g:
```

Comment:
oLed -> OLED.

Also, could you use `isLeftHand` if you wanted the context based on the hand of the keyboard rather than which one is the master?

Reply:
Yes, please do that. has_usb is also AVR specific.

In split_util.c there's is_keyboard_left() and is_keyboard_master() functions specifically to identify these states, ideally those would be used instead of peeking into global variables.

Reply:
`split_util.h` currently only has the declaration for the variable, not the functions, so those would need to be added first, right?

Reply:
Yeah, that definitely should have been done, by me, previously. 🤦‍♂️

Reply:
Fixed in the docs, will wait for your pr to land before I expose those functions to mine if necessary.
Also just noticed the `matrix_master_OLED_init` function in there that I'll need to do something with as it no longer fits with the new driver in that location.